src/js/libgeo/GeoOps.js

```diff
@@ -2349,6 +2351,60 @@ geoOps._helper.conicOtherIntersection = function(conic, a, b) {
     return res;
 };
 
+geoOps.Dist = {};
+geoOps.Dist.kind = "V";
+geoOps.Dist.signature = ["P", "P"];
+geoOps.Dist.updatePosition = function(el) {
+    var a = csgeo.csnames[el.args[0]].homog;
+    var b = csgeo.csnames[el.args[1]].homog;
+    el.value = List.abs(List.sub(List.normalizeZ(a), List.normalizeZ(b)));
+};
+
+geoOps.Angle = {};
+geoOps.Angle.kind = "V";
+geoOps.Angle.signature = ["L", "L", "P"];
+geoOps.Angle.initialize = function(el) {
+    if (el.angle === undefined)
+        el.angle = 0.5 * Math.PI;
+    putStateComplexNumber(CSNumber._helper.input(el.angle));
+};
+geoOps.Angle.updatePosition = function(el) {
+    var a = csgeo.csnames[el.args[0]].homog;
+    var b = csgeo.csnames[el.args[1]].homog;
+    var p = csgeo.csnames[el.args[2]].homog;
+    var ap = List.cross(a, List.linfty);
+    var bp = List.cross(b, List.linfty);
+    var cr = List.crossratio3(ap, bp, List.ii, List.jj, p);
+    var ang = CSNumber.mult(CSNumber.complex(0, 0.5), CSNumber.log(cr));
+    var prev = getStateComplexNumber();
+    var diff = (prev.value.real - ang.value.real) / Math.PI;
+    var winding = Math.round(diff);
+    if (!tracingInitial && Math.abs(winding - diff) > 1e-2)
+        requestRefinement();
+    ang = CSNumber.complex(winding * Math.PI + ang.value.real, ang.value.imag);
+    putStateComplexNumber(ang);
+    el.value = General.withUsage(ang, "Angle");
+};
+geoOps.Angle.stateSize = 2;
+
+geoOps.TextImpl = {};
+geoOps.TextImpl.kind = "Text";
+geoOps.TextImpl.signature = [];
+geoOps.TextImpl.updatePosition = noop;
+geoOps.TextImpl.initialize = function(el) {
+    el.text = String(el.text);
+    el.size = CSNumber.real(el.size ? +el.size : defaultAppearance.textsize);
+    if (el.pos) el.pos = geoOps._helper.initializePoint(el);
+    if (el.dock) {
+        if (el.dock.offset && el.dock.offset.length === 2)
+            el.dock.offset = List.realVector([+el.dock.offset[0], +el.dock.offset[1]]);
+        else
+            el.dock.offset = List.realVector([0, 0]);
+    }
+};
+
+function noop() {}
```

Comment:
[No Operation](https://en.wikipedia.org/wiki/NOP). Would you prefer `nop`, or spelling it out as `noOperation`?

It's used for `geoOps.TextImpl.updatePosition` since text position isn't dealt with during tracing, only when rendering. But once we have a function of this name, we could use it in other locations as well \[[1](https://github.com/CindyJS/CindyJS/blob/89219dec8d71a3df8fa1ee74b8e1821acf2a1ace/src/js/libgeo/Tracing.js#L377),[2](https://github.com/CindyJS/CindyJS/blob/89219dec8d71a3df8fa1ee74b8e1821acf2a1ace/src/js/libgeo/Tracing.js#L461),[3](https://github.com/CindyJS/CindyJS/blob/89219dec8d71a3df8fa1ee74b8e1821acf2a1ace/src/js/libgeo/Tracing.js#L715),[4](https://github.com/CindyJS/CindyJS/blob/89219dec8d71a3df8fa1ee74b8e1821acf2a1ace/src/js/Setup.js#L119),[5](https://github.com/CindyJS/CindyJS/blob/89219dec8d71a3df8fa1ee74b8e1821acf2a1ace/src/js/Head.js#L14)\].